Nil pointer exception when creating sr

### Summary 

There is an uncontrolled nil pointer exception that occurs if the creation of a SourceRepository fails.

### Steps to reproduce the behavior

Try to create a jx environments with a very long name.

### Expected behavior

Either the environment is created or an error is reported.

### Actual behavior

Nil pointer exception:

```
Creating 01--st-romen-in-romen-rome-termediate-159172-721182 Environment in namespace **
Created environment 01--st-romen-in-romen-rome-termediate-159172-721182
Namespace 01--st-romen-in-romen-rome-termediate-159172-721182 created
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x2ebf59b]

goroutine 1 [running]:
github.com/jenkins-x/**/pkg/cmd/create.(*CreateEnvOptions).RegisterEnvironment(0xc000341800, 0xc000829400, 0x58dd700, 0xc0000341c0, 0x5857e20, 0xc000d751e0, 0x2, 0x0)
	*****************/pkg/cmd/create/create_env.go:323 +0x41b
github.com/jenkins-x/**/pkg/cmd/create.(*CreateEnvOptions).Run(0xc000341800, 0x0, 0x0)
	*****************/pkg/cmd/create/create_env.go:254 +0xbfa
github.com/jenkins-x/**/pkg/cmd/create.NewCmdCreateEnv.func1(0xc00078a780, 0xc00074e900, 0x0, 0xc)
	*****************/pkg/cmd/create/create_env.go:96 +0x6f
github.com/spf13/cobra.(*Command).execute(0xc00078a780, 0xc00074e840, 0xc, 0xc, 0xc00078a780, 0xc00074e840)
	/workspace/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:830 +0x2ae
github.com/spf13/cobra.(*Command).ExecuteC(0xc0003c2280, 0xc000703200, 0x57fa200, 0xc0000bc000)
	/workspace/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914 +0x2fc
github.com/spf13/cobra.(*Command).Execute(...)
	/workspace/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
github.com/jenkins-x/**/cmd/**/app.Run(0x0, 0x0, 0x0, 0xc000723f88, 0xc0000a0058)
	*****************/cmd/**/app/**.go:19 +0x185
main.main()
	*****************/cmd/**/**.go:11 +0x32
```

### Jx version

The output of `jx version` is:

```
NAME               VERSION
jx                 2.0.1192+cjxd.7
Kubernetes cluster v1.14.10-gke.36
kubectl            v1.14.7
git                2.21.1 (Apple Git-122.3)
Operating System   Mac OS X 10.15.5 build 19F101

```

--- a/pkg/cmd/create/create_env.go
+++ b/pkg/cmd/create/create_env.go
@@ -320,7 +320,13 @@ func (o *CreateEnvOptions) RegisterEnvironment(env *v1.Environment, gitProvider 
 				return err
 			}
 			sr, err := kube.GetOrCreateSourceRepository(jxClient, devNs, gitInfo.Name, gitInfo.Organisation, gitInfo.HostURLWithoutUser())
-			log.Logger().Debugf("have SourceRepository: %s\n", sr.Name)
+			if err != nil {
+				return errors.Wrap(err, "getting or creating source repository")
+			} else if sr == nil {
+				log.Logger().Warnf("unexpected returned nil value for sr in GetOrCreateSourceRepository")
+			} else {
+				log.Logger().Debugf("have SourceRepository: %s\n", sr.Name)
+			}
 
 			err = o.GenerateProwConfig(devNs, devEnv)
 			if err != nil {

